dotCMS/src/main/java/com/dotcms/util/pagination/CategoriesPaginator.java

```diff
@@ -63,4 +94,42 @@ public PaginatedArrayList<Category> getItems(final User user, final String filte
             throw new DotRuntimeException(e);
         }
     }
+
+    /**
+     * Search for categories at every level.
+     *
+     * @param user to check Permission
+     * @param searchingCriteria Criteria for Searching
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    private PaginatedCategories searchAllLevels(final User user,
+                                                final CategoryFactory.CategorySearchCriteria searchingCriteria)
+            throws DotDataException, DotSecurityException {
+
+        return categoryAPI.findAll(searchingCriteria, user, false);
+    }
+
+    /**
+     * Search for a category at a single level. This could be either the top level or within the immediate children of any category.
+     *
+     * @param user to Check permission
+     * @param searchingCriteria Search Criteria
+
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    private PaginatedCategories searchInOneLevel(final User user,
+                                                 final CategoryFactory.CategorySearchCriteria searchingCriteria,
```

Comment:
I think the CategorySearch should be not an inner class to avoid an unnecessary coupling

Reply:
done https://github.com/dotCMS/core/pull/29091/commits/9b6545575567467dd085b7ceb589e63b2659c293#diff-357d4ceb69b362977138b8a7e1ac932d6344e23e1dc5b5fccf4d4fab851ffc95R16